frontends/web/src/routes/account/add/add.tsx

```diff
@@ -57,81 +61,99 @@ class ManageAccount extends Component<Props, State> {
                 const onlyOneSupportedCoin = (coins.length === 1);
                 this.setState({
-                    // @ts-ignore
-                    coinCode: onlyOneSupportedCoin ? coins[0] : 'choose',
-                    supportedCoins: coins
+                    coinCode: onlyOneSupportedCoin ? coins[0].coinCode : 'choose',
+                    step: onlyOneSupportedCoin ? 'choose-name' : 'select-coin',
+                    supportedCoins: coins,
                 });
-            })
-            .catch(console.error);
-    }
-
-    private getStep = () => {
-        const { step } = this.state;
-        if (this.onlyOneSupportedCoin()) {
-            return step === 0 ? 'choose-name' : 'success';
-        }
-        switch (step) {
-            case 0: return 'select-coin';
-            case 1: return 'choose-name';
-            case 2: return 'success';
-        }
+            });
     }
 
     private back = () => {
-        this.setState(({ step }) => {
-            switch (step) {
-                case 0:
-                case 1: return ({ step: 0 });
-                case 2: return ({ step: 1 });
-            }
-        });
+        switch (this.state.step) {
+            case 'choose-name':
+                this.setState({ step: 'select-coin' });
+                break;
+            case 'success':
+                this.setState({ step: 'choose-name' });
+                break;
+        }
     }
 
-    private next = () => {
-        const { accountName, coinCode } = this.state;
-        switch (this.getStep()) {
+    private next = (e: Event) => {
+        e.preventDefault();
+        const { accountCode, accountName, coinCode, step } = this.state;
+        const { t } = this.props;
+        switch (step) {
             case 'select-coin':
-                console.info(`${coinCode} selected`);
-                this.setState({ step: 1 });
+                this.setState({ step: 'choose-name' });
                 break;
             case 'choose-name':
                 interface ResponseData {
                     success: boolean;
+                    accountCode?: string;
                     errorCode?: 'alreadyExists' | 'limitReached';
                     errorMessage?: string;
                 }
 
                 apiPost('account-add', {
                     coinCode,
                     name: accountName,
-                }).then((data: ResponseData) => {
-                    if (data.success) {
-                        this.setState({ step: 2 });
-                        //route('/account/' + data.accountCode);
-                    } else {
-                        if (data.errorCode) {
-                            alertUser(this.props.t(`addAccount.error.${data.errorCode}`));
+                })
+                    .then((data: ResponseData) => {
+                        if (data.success) {
+                            this.setState({
+                                accountCode: data.accountCode,
```

Comment:
why this and not route right here, like the commented line below?

Reply:
User should see the success message and then when clicking primary button on success step it should go to the account.

Else it would go directly to account without showing the success step.